src/tools/schema2graph.ts

```diff
@@ -60,13 +54,21 @@ export class SchemaGraph {
   nodes: SchemaNode[] = [];
   startNodes: SchemaNode[];
   internalClassIndex = 0;
-
-  constructor(readonly particleSpec: ParticleSpec) {
+  nameGenerator: (node: SchemaNode, i: number)  => string;
+  aliasGenerator: (node: SchemaNode) => string[];
+
+  constructor(
+    readonly particleSpec: ParticleSpec,
+    nameGenerator: (node: SchemaNode, i: number)  => string,
+    aliasGenerator: (node: SchemaNode)  => string[]
+    ) {
+    this.nameGenerator = nameGenerator;
+    this.aliasGenerator = aliasGenerator;
```

Comment:
Since this class is already mixing parameter properties with other, non-ctor-initialised members, you might as well follow suit:
```
constructor(
  readonly particleSpec: ParticleSpec,
  private nameGenerator: (node: SchemaNode, i: number)  => string,
  private aliasGenerator: (node: SchemaNode)  => string[]
) {
```